aviary/mission/height_energy_problem_configurator.py

```diff
@@ -0,0 +1,625 @@
+import numpy as np
+
+import openmdao.api as om
+
+from dymos.transcriptions.transcription_base import TranscriptionBase
+
+from aviary.mission.flops_based.phases.build_landing import Landing
+from aviary.mission.flops_based.phases.build_takeoff import Takeoff
+from aviary.mission.flops_based.phases.energy_phase import EnergyPhase
+from aviary.mission.phase_builder_base import PhaseBuilderBase
+from aviary.mission.problem_configurator import ProblemConfiguratorBase
+from aviary.utils.functions import wrapped_convert_units
+from aviary.utils.process_input_decks import initialization_guessing
+from aviary.variable_info.enums import AnalysisScheme, LegacyCode
+from aviary.variable_info.variables import Aircraft, Mission, Dynamic, Settings
+from aviary.subsystems.propulsion.utils import build_engine_deck
+
+if hasattr(TranscriptionBase, 'setup_polynomial_controls'):
+    use_new_dymos_syntax = False
+else:
+    use_new_dymos_syntax = True
+
+
+class HeightEnergyProblemConfigurator(ProblemConfiguratorBase):
+    """
+    A Height-Energy specific builder that customizes AviaryProblem() for use with
+    height energy phases.
+    """
+
+    def initial_guesses(self, prob):
```

Comment:
We have both `initial_guesses()` (Set any initial guesses for variables in the aviary problem) and `add_guesses()` (Adds the initial guesses for each variable of a given phase to the problem.). I'm wondering if we can consolidate? I'm sure that the reason we have two is because we setup guesses in two places in M4L2

Reply:
It is kind of weird, and something we didn't really notice until we pulled the code out. I think untangling all of these strands is going to be difficult, but will result in more maintainable code.